src/components/NumberInput.tsx

```diff
@@ -0,0 +1,147 @@
+import { useState } from "react";
+import { Button, Stack, TextField } from "@mui/material";
+
+const Modes = {
+  /** Natural numbers from 0 to inf */
+  NATURAL: /^([0-9]+)$/,
+  /** Integers from -inf to inf */
+  INTEGER: /^[+\\-]?([0-9]+)$/,
+  /** Floating point numbers from -inf to inf, accepts values such as 1. and .1 as valid*/
+  FLOATING:
+    /^[+\\-]?(([0-9]+)|([0-9]+[\\.])|([\\.][0-9]+)|([0-9]+[\\.][0-9]+))$/,
+  /** Floating point numbers from -inf to inf, accepts values such as 1.e1 and .1e1 as valid*/
+  SCIENTIFIC:
+    /^[+\\-]?(([0-9]+)|([0-9]+[\\.])|([\\.][0-9]+)|([0-9]+[\\.][0-9]+))([eE][+\\-]?[0-9]+)?$/,
+};
+
+interface NumberInputTextProps {
+  label: string;
+  numberMode: keyof typeof Modes;
+  numberText: string;
+  setNumberText: (v: string) => void;
+  isValid: boolean;
+  setIsValid: (v: boolean) => void;
+  handleSubmit?: () => void;
```

Comment:
Was the submit functionality in comms requirements? I feel like a NumberInput should just allow input and display of numbers really well. Submission is a different concern and doesn't belong in the same abstraction. I know you were following the Visit component from sci-react-ui, but likewise I feel it's wrong for the submit button to be there.